packages/terra-table/src/Table.jsx

```diff
@@ -0,0 +1,353 @@
+   * a unique id.
+   */
+  id: PropTypes.string.isRequired,
+
+  /**
+   * @private
+   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
+  */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+
+  /**
+      * Data for content in the body of the table. Rows will be rendered in the order given.
+      */
+  rows: PropTypes.arrayOf(rowShape).isRequired,
+
+  /**
+   * String that identifies the element (or elements) that labels the table.
+   */
+  ariaLabelledBy: PropTypes.string,
+
+  /**
+   * String that labels the table for accessibility. If ariaLabelledBy is specified, ariaLabel will not be used.
+   */
+  ariaLabel: PropTypes.string,
+
+  /**
+   * Data for pinned columns. Pinned columns are the stickied leftmost columns of the table.
+   * Columns will be presented in the order given.
+   */
+  pinnedColumns: PropTypes.arrayOf(columnShape),
+
+  /**
+   * Data for overflow columns. Overflow columns are rendered in the table's horizontal overflow.
+   * Columns will be presented in the order given.
+   */
+  overflowColumns: PropTypes.arrayOf(columnShape),
+
+  /**
+   * Number indicating the default column width in px. This value will be used if no overriding width value is provided on a per-column basis.
+   */
+  defaultColumnWidth: PropTypes.number,
+
+  /**
+   * String that specifies the column height. Any valid CSS height value is accepted.
+   */
+  columnHeaderHeight: PropTypes.string,
+
+  /**
+   * String that specifies the height for the rows in the table. Any valid CSS value is accepted.
+   */
+  rowHeight: PropTypes.string,
+
+  /**
+   * Number indicating the index of the column that represents row header. Index is 0 based and cannot exceed one less than the number of columns in the table.
+   */
+  rowHeaderIndex: validateRowHeaderIndex,
+
+  /**
+   * Function that is called when a resizable column is resized. Parameters:
+   * @param {string} columnId columnId
+   * @param {string} requestedWidth requestedWidth
+   */
+  onColumnResize: PropTypes.func,
+
+  /**
+   * Callback function that is called when a selectable cell is selected. Parameters:
+   * @param {string} rowId rowId
+   * @param {string} columnId columnId
+   */
+  onCellSelect: PropTypes.func,
+
+  /**
+   * Callback function that is called when a selectable column is selected. Parameters:
+   *  @param {string} columnId columnId
+   */
+  onColumnSelect: PropTypes.func,
+
+  /**
+   * Boolean indicating whether or not the DataGrid should allow entire rows to be selectable. An additional column will be
+   * rendered to allow for row selection to occur.
+   */
+  hasSelectableRows: PropTypes.bool,
+};
+
+const defaultProps = {
+  rowHeaderIndex: 0,
+  defaultColumnWidth: 200,
+  columnHeaderHeight: '2.5rem',
+  rowHeight: '2.5rem',
+  pinnedColumns: [],
+  overflowColumns: [],
```

Comment:
Should we add a default empty array for `rows` as well?

Reply:
Right now I'm requiring `rows` to be supplied (but it can be empty) since in DataGrid this is actually a bug right now (component will not render otherwise). But I could also see a case where we might want to default to `[]` if `rows` is not specified.